linux/Jenkinsfile

```diff
@@ -338,11 +387,34 @@ def uploadDebArtifacts(String buildArch) {
         } else {
             env.BUILDARCH = debArchList[buildArch]
             env.DISTROLIST = distro_list
-            jf 'rt u **/build/ospackage/temurin-*${BUILDARCH}.deb deb/pool/main/t/temurin-${VERSION}/ --target-props=${DISTROLIST}deb.component=main;deb.architecture=${BUILDARCH} --flat=true'
-            break
+
+            def PackFiles = findFiles(glob: "**/build/ospackage/temurin-*${debArchList[buildArch]}.deb") // List All Packages To Upload
+            for (PackFile in PackFiles) {
+              def FileName = PackFile.name
+              def Target = "https://adoptium.jfrog.io/artifactory/deb/pool/main/t/temurin-${VERSION}/${FileName}"
+
+              echo "File : ${PackFile}"
+              echo "Filename : ${FileName}"
+              echo "Target : ${Target}"
+
+              try {
+                def ResponseCode = sh(script: "curl -o /dev/null --silent --head --write-out '%{http_code}' ${Target}", returnStdout: true).trim()
```

Comment:
we should use the JFrog CLI to check if a file exists rather than writing a custom curl script. Check out how I do this for the cacerts upload script:

https://github.com/adoptium/installer/blob/master/.github/workflows/cacert-publish.yml#L43-L52

```bash
if jf rt s "deb/pool/main/a/adoptium-ca-certificates/$(basename $FILE)" > /dev/null; then
  echo file_exists=true >> "$GITHUB_OUTPUT"
else
  echo file_exists=false >> "$GITHUB_OUTPUT"
fi
```

Reply:
Ah, I had several attempts at this, but couldn't get it to work correctly, I'll try it using this method, see if I have any more success...

Reply:
In Jenkins pipeline format it should look something like this:

```groovy
 jf 'rt s ${Target}'
```

Reply:
Having spent the day, trying many different varieties, I think this is an issue or a lack of feature in the jfrog cli plugin for jenkins, the command line/shell driven normal version can be handled in the usual fashion, however the jenkins plugin, other than outputting test to the screen, does not provide anything that can be processed, so Im reverting this to the CURL based check , and will log issues with the provider, with the idea that this can be improved with a future update of the plugin.